### ISSUE TITLE
[Bug Report] No way to emit v-text-field value when using masks

### ISSUE THREAD

**Issue by neanderthil** [vuetifyjs/vuetify#8192]
### Environment
**Vuetify Version:** 2.0.2
**Last working version:** 1.5.0-1.5.0-beta.1.0
**Vue Version:** 2.6.10
**Browsers:** Chrome 75.0.3770.142
**OS:** Linux x86_64

### Steps to reproduce
Use the example in docs

### Expected Behavior
Value can be emitted without the mask.

### Actual Behavior
Option to emit value.

### Reproduction Link
<a href="https://codepen.io/neanderthil/pen/ZgKGKj" target="_blank">https://codepen.io/neanderthil/pen/ZgKGKj</a>

### Other comments
This used to work before the 2.0 upgrade with the built-in masks. With the removal of those in 2.0 the 3rd party library directive solution is inadequate.

Possible solution would be a slot for custom input text where we could use the vue-the-mask component <mask> rather than be limited to just the directive, which you can emit the value with. So far I have found no way to get just the value after hours of research. This breaks quite a few things in my application as I am validating on types.

Thanks

**Comment by diazemiliano** [comment]
Related to PR: https://github.com/vuetifyjs/vuetify/pull/7304

**Comment by nekosaur** [comment]
vue-the-mask exports its masker function. so if you need the unmasked value you can do

```js
import { masker } from 'vue-the-mask'
...
const unmasked = masker(maskedValue, mask, false)
```

earlier I did experiment with adding a way to do this

```
<v-text-field v-mask="mask" v-model="value" @unmasked="...">
```

by check `vnode.data.on.unmasked` in the directive and calling it with the unmasked value. it works, but not sure if it would be of much greater use than manually unmasking when you need to.

**Comment by deraw** [comment]
Hi @nekosaur!
I think the removal of mask functionality should be mentionned somewhere on the [release notes of v2.0](https://github.com/vuetifyjs/vuetify/releases/tag/v2.0.0). This is a breaking change for some user and it's not documented!
Thanks

**Comment by nekosaur** [comment]
It is definitely a breaking change and should be mentioned. I has been added to release notes.

**Comment by aldarund** [comment]
@nekosaur are u sure it exports masker? I dont really see. I see only

```
{                                                                          
  TheMask: [Getter],
  mask: [Getter],
  tokens: [Getter],
  version: [Getter],
  default: [Function: r]
}
```

**Comment by nekosaur** [comment]
Sorry, I was apparently looking at a fork (https://github.com/RonaldJerez/vue-the-mask). Perhaps we should try to find a better alternate 3rd party library since vue-the-mask has been pretty dead for more than a year now.

**Comment by vincentauger** [comment]
@nekosaur Stumble onto this breaking change today, tried the vue-the-mask route but ran into the same problem with model being masked. Tried with v-mask (https://github.com/probil/v-mask) as plug-in and work as intended.

**Comment by nekosaur** [comment]
Good to know. I'll try it out and see if I can update the documentation to use that instead of vue-the-mask

**Comment by aldarund** [comment]
last time i tried probil/v-mask when vuetify 2 was in beta it wasnt working for vuetify, good that its fixed then :)

**Comment by nekosaur** [comment]
@vincentauger I just looked at v-mask, and it seems to function the same as vue-the-mask? i.e. it does not allow you to get the unmasked value

**Comment by vincentauger** [comment]
@nekosaur You are correct, I made a silly mistake after trying too many work around last night and left a computed values that manually unmasked it in my tests, sorry about this. I'll keep searching here.

**Comment by egeersoz** [comment]
Why was the mask feature (which I consider critical for UX, as well as keeping incoming data clean) removed and delegated to third-party libraries? It just adds another dependency to worry about.

**Comment by KaelWD** [comment]
![image](https://user-images.githubusercontent.com/16421948/62830633-33465480-bc55-11e9-9045-4f422cc25e30.png)

**Comment by egeersoz** [comment]
I understand, but this caught a lot of people off guard. Based on the chat log you posted, you guys discussed this internally and made a decision to remove the feature. It would have been great if you had first solicited comments from the actual users though. You would have gathered valuable feedback, and who knows, someone may even have stepped up to maintain the feature moving forward.

In addition, I think in these situations you should first make sure that the other options that are available have feature parity before removing the feature from Vuetify. Otherwise you're simply passing the aforementioned technical debt to your users. In this particular case, none of the third-party libraries out there seem to emit the unmasked value (I think I tried six or seven before giving up). This means the unmasking needs to be performed manually post-emit, which gets real messy.

I love the work you guys are doing, and Vuetify in general is fantastic. 2.0 especially has been a total slam dunk. <3 But when it comes to feature removals, some sort of standard operating procedure, developed in collaboration with the community, would be great to have.

**Comment by KaelWD** [comment]
Looks like this particular issue was actually brought up by one user during beta, but nobody really complained further about it. We do try to get feedback where possible, but it's hard to tell how people are actually using features if at all.
I don't think we're actually opposed to adding masking back, but it would definitely have to be done in a more maintainable way. The previous implementation was done by a community member and made it difficult to fix other text field problems due to how tightly coupled it was to everything. 

![image](https://user-images.githubusercontent.com/16421948/62831065-b919ce00-bc5c-11e9-858f-6a6f73bceece.png)

**Comment by neanderthil** [comment]
Does anyone have a workaround at least? I'm really not looking to manage this manually.

**Comment by vincentauger** [comment]
Currently, I’m unmasking using regex and a computed value as a work-around.

For example:
 
`unmaskedPhone() {
            return this.user.phone_number.replace(/\D+/g, "")
        }`

**Comment by horaciosolorio** [comment]
How to remove v-mask directive for custom component, i need to don't mask the input some times but if i send v-mask="" don´t work any idea ??

**Comment by horaciosolorio** [comment]
> How to remove v-mask directive for custom component, i need to don't mask the input some times but if i send v-mask="" don´t work any idea ??

Hello any update for this issue i didn't recibe any feed back could you help me please

**Comment by paulodiogo** [comment]
> I understand, but this caught a lot of people off guard. Based on the chat log you posted, you guys discussed this internally and made a decision to remove the feature. It would have been great if you had first solicited comments from the actual users though. You would have gathered valuable feedback, and who knows, someone may even have stepped up to maintain the feature moving forward.
> 
> In addition, I think in these situations you should first make sure that the other options that are available have feature parity before removing the feature from Vuetify. Otherwise you're simply passing the aforementioned technical debt to your users. In this particular case, none of the third-party libraries out there seem to emit the unmasked value (I think I tried six or seven before giving up). This means the unmasking needs to be performed manually post-emit, which gets real messy.
> 
> I love the work you guys are doing, and Vuetify in general is fantastic. 2.0 especially has been a total slam dunk. <3 But when it comes to feature removals, some sort of standard operating procedure, developed in collaboration with the community, would be great to have. Not just a "we will mention this in the migration documentation" but a more inclusive discussion with the community.

I'm terrified with this, remove a feature without asking the community. I don't even think migrate version 2 now. I had many issues doing the decimal mask, the mask that came with vuetify was just perfect for the other cases. I think this is a great lost for the project.

**Comment by rammie** [comment]
Just wanted to point out that this causes the mask + counter used together to not work as expected with any of the 3rd party masking libraries (vue-the-mask, vue-mask and vue-mask-directive (this was working correctly in vuetify 1.X with the builtin mask feature).

**Comment by GsHeri** [comment]
would it be possible to release the feature as an optional component? so that users who want to use it, can jsut drop it in their project at their own risk? maybe customize it or something? as far as i've read, none of the alternatives returns unmasked values. the vuetify one did, so maybe someone could release it on its own? i'm not sure if i'd want (or could) write a solid replacement myself

**Comment by johnleider** [comment]
Will be resolved as part of #10890

If you have any questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).
